server/channels/app/slashcommands/command_invite.go

```diff
@@ -59,28 +70,127 @@ func (i *InviteProvider) doCommand(a *app.App, c request.CTX, args *model.Comman
+					differentChannels = append(differentChannels, targetUser.Username)
+				}
+			} else if userError == UserNotInTeam {
+				nonTeamUsers = append(nonTeamUsers, targetUser.Username)
+			} else if userError == IsConstrained {
+				teamConstrained = append(teamConstrained, targetUser)
+			} else if userError == UserInChannel {
+				usersInChannel = append(usersInChannel, targetUser.Username)
+			} else {
+				errorUsers = append(errorUsers, targetUser)
 			}
-			if args.ChannelId != targetChannel.Id {
-				*resps = append(*resps, args.T("api.command_invite.success", map[string]any{
-					"User":    targetUser.Username,
-					"Channel": targetChannel.Name,
-				}))
+		}
+	}
+
+	if len(usersInChannel) > 0 {
+		if len(usersInChannel) > 10 {
+			*resps = append(*resps,
+				args.T("api.command_invite.user_already_in_channel.overflow", map[string]any{
+					"FirstUser": "@" + usersInChannel[0],
+					"Others":    len(usersInChannel) - 1,
+				}),
+			)
+		} else {
+			usersString := map[string]any{
+				"User": "@" + strings.Join(usersInChannel, ", @"),
+			}
+			*resps = append(*resps,
+				args.T("api.command_invite.user_already_in_channel.app_error", len(usersInChannel), usersString),
+			)
+		}
+	}
+
+	if len(differentChannels) > 0 {
+		if len(differentChannels) > 10 {
+			*resps = append(*resps,
+				args.T("api.command_invite.successOverflow", map[string]any{
+					"FirstUser": "@" + differentChannels[0],
+					"Others":    len(differentChannels) - 1,
+					"Channel":   "",
+				}),
+			)
+		} else {
+			usersString := map[string]any{
+				"Users":   "@" + strings.Join(differentChannels, ", @"),
+				"Channel": "test",
+			}
+			*resps = append(*resps,
+				args.T("api.command_invite.success", usersString),
+			)
+		}
+	}
+
+	if len(nonTeamUsers) > 0 {
+		if len(nonTeamUsers) > 10 {
+			*resps = append(*resps,
+				args.T("api.command_invite.user_not_in_team.messageOverflow", map[string]any{
+					"FirstUser": "@" + nonTeamUsers[0],
+					"Others":    len(nonTeamUsers) - 1,
+					"Team":      "",
+				}),
+			)
+		} else {
+			usersString := map[string]any{
+				"Users": "@" + strings.Join(nonTeamUsers, ", @"),
+				"Team":  "test",
 			}
+			*resps = append(*resps,
+				args.T("api.command_invite.user_not_in_team.app_error", usersString),
+			)
 		}
 	}
 
+	if len(teamConstrained) > 0 {
+		*resps = append(*resps,
+			args.T("api.command_invite.group_constrained_user_denied"),
+		)
+	}
+
+	if len(errorUsers) > 0 {
+		*resps = append(*resps,
+			args.T("api.command_invite.fail.app_error"),
+		)
+	}
+
 	if len(*resps) > 0 {
 		return strings.Join(*resps, "\n")
 	}
-
 	return ""
 }
 
+func (i *InviteProvider) getUsersFromMentionName(a *app.App, mentionName string) []*model.User {
+	userProfile, err := a.Srv().Store().User().GetByUsername(mentionName)
+	if err == nil && userProfile.DeleteAt == 0 {
+		return []*model.User{userProfile}
+	}
+
+	group, appErr := a.GetGroupByName(mentionName, model.GroupSearchOpts{})
```

Comment:
Do we need to use `FilterAllowReference` here?

Reply:
Yep...I believe so....added.